modules/base/src/algorithm/mesh/meshclipping.cpp

```diff
@@ -0,0 +1,572 @@
+                            return [outRam](const std::vector<uint32_t>& indices,
+                                            const std::vector<float>& weights,
+                                            std::optional<vec3> position, std::optional<vec3>) {
+                                outRam->add(position ? *position : mix(*outRam, indices, weights));
+                                return static_cast<uint32_t>(outRam->getSize() - 1);
+                            };
+                        } else if (bufferType == BufferType::NormalAttrib) {
+                            return [outRam](const std::vector<uint32_t>& indices,
+                                            const std::vector<float>& weights, std::optional<vec3>,
+                                            std::optional<vec3> normal) {
+                                outRam->add(normal ? *normal : mix(*outRam, indices, weights));
+                                return static_cast<uint32_t>(outRam->getSize() - 1);
+                            };
+                        }
+                    }
+
+                    if constexpr (DataFormat<ValueType>::numtype == NumericType::Float) {
+                        return [outRam](const std::vector<uint32_t>& indices,
+                                        const std::vector<float>& weights, std::optional<vec3>,
+                                        std::optional<vec3>) {
+                            outRam->add(mix(*outRam, indices, weights));
+                            return static_cast<uint32_t>(outRam->getSize() - 1);
+                        };
+                    } else {  // Only interpolate floating point buffers;
+                        return [outRam](const std::vector<uint32_t>& indices,
+                                        const std::vector<float>& weights, std::optional<vec3>,
+                                        std::optional<vec3>) {
+                            const auto it = std::max_element(weights.begin(), weights.end());
+                            const auto index = std::distance(weights.begin(), it);
+
+                            outRam->add(static_cast<ValueType>((*outRam)[indices[index]]));
+                            return static_cast<uint32_t>(outRam->getSize() - 1);
+                        };
+                    }
+                });
+        interpolateFunctors.push_back(functor);
+    }
+
+    const detail::InterpolateFunctor addInterpolatedVertex =
+        [&interpolateFunctors](const std::vector<uint32_t>& indices,
+                               const std::vector<float>& weights, std::optional<vec3> position,
+                               std::optional<vec3> normal) -> uint32_t {
+        uint32_t res = 0;
+        for (auto& fun : interpolateFunctors) res = fun(indices, weights, position, normal);
+        return res;
+    };
+
+    if (!posBuffer) {
+        throw Exception("Unsupported mesh type, vec3 position buffer not found",
```

Comment:
hm, that could be an issue. There are some 2D meshes around as well. Guess that could be a separate PR if needed.

Reply:
clipping in 2D is a very different problem....